collated/test/jeffreygohkw.md

```diff
@@ -0,0 +1,861 @@
+        assertParseFailure(parser, "phone",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, "email",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, "address",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+
+        // no field or order
+        assertParseFailure(parser, "random text",
+                String.format(MESSAGE_INVALID_INPUT, SortCommand.MESSAGE_USAGE));
+
+    }
+
+
+    @Test
+    public void parse_validArguments_success() throws ParseException {
+        SortCommand expectedCommand;
+        SortCommand actualCommand;
+
+        expectedCommand = new SortCommand("name", "asc");
+        actualCommand = parser.parse("name asc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("name", "desc");
+        actualCommand = parser.parse("name desc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("phone", "asc");
+        actualCommand = parser.parse("phone asc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("phone", "desc");
+        actualCommand = parser.parse("phone desc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("email", "asc");
+        actualCommand = parser.parse("email asc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("email", "desc");
+        actualCommand = parser.parse("email desc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("address", "asc");
+        actualCommand = parser.parse("address asc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
+
+        expectedCommand = new SortCommand("address", "desc");
+        actualCommand = parser.parse("address desc");
+
+        assertEquals(expectedCommand.getField(), actualCommand.getField());
+        assertEquals(expectedCommand.getOrder(), actualCommand.getOrder());
```

Comment:
Try to minimize repetitive code blocks by extracting out similarities as utility methods.